civihr_employee_portal/js/reports.js

```diff
@@ -926,6 +926,74 @@
             }.bind(this)
         });
 
+        $.ajax({
+            type: 'GET',
+            url: this.getJsonUrl() + ( typeof path !== 'undefined' ? path : '' ),
+            success: function(data) {
+                var mappedResults = normaliseSummaryData(data.results);
+
+                replaceSummaryTable(mappedResults, this.$DOM);
+            }.bind(this)
+        });
+
+        function normaliseSummaryData(summary) {
+            var normalisedSummary = {},
+                i;
+
+            // The data can have two formats, depending on particular report:
+            // data = { results: [{ departament: "xxx", count: 1 }, ...]) }
+            // or
+            // data = { results: [{ gender: "aaa" }, { gender: "aaa" }, ...] }
+            //
+            // The code inside if takes care of the first possibility
+            // The else takes care of the second
+            if (summary[0] !== undefined && summary[0].data.count !== undefined) {
+                for(i in summary) {
+                    normalisedSummary[summary[i].data.department] = +summary[i].data.count;
+                }
+            } else {
+                normalisedSummary = summary.map(function (row) {
```

Comment:
Super-terse alternative (to be checked, not 100% sure :smile:)
```js
normalisedSummary = summary.reduce(function (accumulator, row) {
    var key = row.data.gender;

    (typeof accumulator[key] === undefined) && (accumulator[key] = 0)

    return ++accumulator[key] && accumulator;
}, {});
```
but the point is that it doesn't seem you really need that `map` intermediate step, and also you are always incrementing by 1, so what's point of having `x:1`?